source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -270,6 +275,43 @@ void AuthenticatorImpl::verifyKey() {
   handleGoodJwt(/*cache_hit=*/false);
 }
 
+void AuthenticatorImpl::addJWTClaimToHeader(const std::string& claim_name,
+                                            const std::string& header_name) {
+  StructUtils payload_getter(jwt_->payload_pb_);
+  const ::Envoy::ProtobufWkt::Value* claim_value;
+  const auto status = payload_getter.GetValue(claim_name, claim_value);
+  std::string str_claim_value;
+  if (status == StructUtils::OK) {
+    if (claim_value->kind_case() == Envoy::ProtobufWkt::Value::kStringValue) {
+      str_claim_value = claim_value->string_value();
+    } else if (claim_value->kind_case() == Envoy::ProtobufWkt::Value::kNumberValue) {
+      if (claim_value->number_value() < 0 ||
+          claim_value->number_value() >=
+              static_cast<double>(std::numeric_limits<uint64_t>::max())) {
+        ENVOY_LOG(debug, "--------claim : {} with value : {} not a valid integer -----------",
+                  claim_name, claim_value->number_value());
+      }
+      const uint64_t int_claim_value = static_cast<uint64_t>(claim_value->number_value());
+      str_claim_value = std::to_string(int_claim_value);
+    } else if (claim_value->kind_case() == Envoy::ProtobufWkt::Value::kBoolValue) {
+      str_claim_value = claim_value->bool_value() ? "true" : "false";
+    } else {
+      ENVOY_LOG(debug,
+                "--------claim : {} is not a primitive type of int, string, or bool -----------",
+                claim_name);
+    }
+    if (!str_claim_value.empty()) {
+      headers_->addCopy(Http::LowerCaseString(header_name), str_claim_value);
+      ENVOY_LOG(debug, "--------claim : {} with value : {} is added to the header : {} -----------",
+                claim_name, str_claim_value, header_name);
+    }
+  } else {
+    ENVOY_LOG(debug,
```

Comment:
Probably should bump some stat - debug logging is not useful in production and we are basically silently failing here.

Reply:
Any followup here?

Reply:
cc : @qiwzhang

Reply:
I feel that adding stat for these cases will not be so useful.  The symptoms of such errors will show up in other ways. In this case, the backend will expect certain claim headers, but due some mistakes, such as claim name typo or wrong type, we will not copy them, the backend will not receive them, it will complain.  They can be easily found in the development test as JWT payload format usually don't change.